src/posthog-core.ts

```diff
@@ -1068,17 +1069,32 @@ export class PostHog {
         return properties
     }
 
+    /**
+     * Add additional set_once properties to the event when creating a person profile. This allows us to create the
+     * profile with mostly-accurate properties, despite earlier events not setting them. We do this by storing them in
+     * persistence.
+     * @param dataSetOnce
+     */
     _calculate_set_once_properties(dataSetOnce?: Properties): Properties | undefined {
         if (!this.persistence || !this._hasPersonProcessing()) {
             return dataSetOnce
         }
+
+        if (this._personProcessingSetOncePropertiesSent) {
+            // We only need to send these properties once. Sending them with later events would be redundant and would
+            // just require extra work on the server to process them.
+            return dataSetOnce
+        }
         // if we're an identified person, send initial params with every event
-        let setOnceProperties = extend({}, this.persistence.get_initial_props(), dataSetOnce || {})
+        const initialProps = this.persistence.get_initial_props()
+        const sessionProps = this.sessionPropsManager?.getSetOnceInitialSessionPropsProps()
+        let setOnceProperties = extend({}, initialProps, sessionProps || {}, dataSetOnce || {})
         const sanitize_properties = this.config.sanitize_properties
         if (sanitize_properties) {
             logger.error('sanitize_properties is deprecated. Use before_send instead')
             setOnceProperties = sanitize_properties(setOnceProperties, '$set_once')
         }
+        this._personProcessingSetOncePropertiesSent = true
```

Comment:
Should this really be set here? I'd prefer we set this once we actually call `capture`. If someone calls `_calculate_set_once_properties` but doesn't actually send the props then we'll miss them

Reply:
Hmm, I liked it here because then it was being read/written in the same function. It's a private function and only called from `capture`. What do you think?

Reply:
I'm afraid it might become a bug in the future, but it does make sense for now. Maybe add a warning at the top of this function saying that it's not a pure function and you should make sure you're sending the results returned from the first ever call?